pos/views.py

```diff
@@ -40,107 +41,108 @@ def login(request):
+                if operation == "cashpayment":
+                    cash.cash_amount += product.price
+                    amount_added += product.price
+                    cash.save()
+
+            order.done = True
+            order.save()
+            order = Order.objects.create(user=request.user)
             succesfully_payed = True
 
-        elif operation == "cardpayment":
-            for product in helper.parse_json_product_list(current_order.order_list):
-                product_in_database = Product.objects.get(product_name=product.product_name)
-                amount_added = amount_added + product.product_price
-                if product_in_database.product_stockApplies:
-                    product_in_database.product_stock = productInDatabase.product_stock - 1
-                    product_in_database.save()
-
-            current_order.order_done = True
-            current_order.save()
-            current_order = Order.objects.create(order_user=request.user.username)
-            succesfully_payed = True
-
-
-
-        else:
-            product_in_database = Product.objects.filter(product_name = operation).first()
-            if product_in_database is not None:
-                parsed_json_list = helper.parse_json_product_list(current_order.order_list)
-
-                i = parsed_json_list.index(product_in_database)
-                del parsed_json_list[i]
-                current_order.order_list = helper.product_list_to_json(parsed_json_list)
-                current_order.order_totalprice = ( current_order.order_totalprice - product_in_database.product_price ).quantize(decimal.Decimal('0.01'))
-
-                if current_order.order_totalprice < 0:
-                    logging.warn("prices below 0! You might be running in to the 10 digit total order price limit")
-                    current_order.order_totalprice = 0
-                current_order.save()
-
-    totalprice = current_order.order_totalprice
-    order_list = helper.parse_json_product_list(current_order.order_list)
-    template = loader.get_template('pos/addition.html')
     context = {
-            'order_list': order_list,
-            'totalprice': totalprice,
+            'order_list': order.productinorder_set.all(),
+            'totalprice': order.total,
             'cash': cash,
             'succesfully_payed': succesfully_payed,
             'payment_error': payment_error,
             'amount_added': amount_added,
     }
-    return HttpResponse(template.render(context, request))
+    return render(request, 'pos/addition.html', context=context)
+
+
+@login_required
+def remove_from_order(request, order_id=None, pio_id=None):
+    """Remove a specific item from the order"""
+
+    # attempt ?pio_id= paramter if it was not in the url
+    if pio_id is None: pio_id = request.GET.get('pio_id', None)
+
+    pio = get_object_or_404(ProductInOrder, id=pio_id)
+    #sanity check order id if it matches, sth was weird with the url otherwise
```

Comment:
Comments should follow `# Sentence.` Capital and space are important for readability. There are other comments that have issues, have a look at the codeclimate report